### PR TITLE
perf(migration): update teams manifest file name in v3 migration

### PR DESCRIPTION
**Pull request by KennethBWSong**
rename from 'manifest.template.json' to 'manifest.template.json'

### CODE REVIEW

File: packages/fx-core/src/core/middleware/utils/appYmlGenerator.ts [hunk 1]
```diff
@@ -112,10 +112,10 @@ export class AppYmlGenerator extends BaseAppYmlGenerator {
       this.handlebarsContext.aadAppName = aadManifest.name;
     }
 
-    const teamsAppManifestPath = path.join(this.projectPath, "appPackage/manifest.template.json");
+    const teamsAppManifestPath = path.join(this.projectPath, "appPackage/manifest.json");
```
Comment (blackchoey) [inline packages/fx-core/src/core/middleware/utils/appYmlGenerator.ts:115]: we may use the `Constants.teamsManifestFileName` constant exported from `projectMigratorV3.ts`

Comment (KennethBWSong) [reply packages/fx-core/src/core/middleware/utils/appYmlGenerator.ts]: Updated, thx!

**Review by blackchoey — APPROVED** [review]

File: packages/fx-core/src/core/middleware/projectMigratorV3.ts [hunk 2]
```diff
@@ -106,6 +106,7 @@ const Constants = {
     flag: "a+",
   },
   envFilePrefix: ".env.",
+  teamsManifestFileName: "manifest.json",
```
Comment (xzf0587) [inline packages/fx-core/src/core/middleware/projectMigratorV3.ts:109]: As the manifest file name is not only used in migration period, move the constant to MetadataV3 in versionMetadata.ts is better.

Comment (KennethBWSong) [reply packages/fx-core/src/core/middleware/projectMigratorV3.ts]: Updated, thx!

**Review by xzf0587 — APPROVED** [review]

**Review by blackchoey — APPROVED** [review]

**Pull request merged**